frontend/src/components/regular/tooltip_button.vue

```diff
@@ -109,148 +102,115 @@ export default Vue.extend( {
       default: null
     },
     'left':{
-      default: false,
-      type: Boolean
+      type: Boolean,
+      default: false
     },
     'button_color': {
+      type: String,
       default: null
     },
     'icon': {
-      default: 'mdi-lifebuoy',
-      type: String
+      type: String,
+      default: 'mdi-lifebuoy'
     },
     'tooltip_message': {
-      default: null,
-      type: String
+      type: String,
+      default: null
     },
     'icon_style': {
+      type: Boolean,
       default: false
      },
-    'data_cy': {
-      default: false
-    },
     'text_style': {
+      type: Boolean,
       default: false
     },
     'active': {
+      type: Boolean,
       default: false
     },
-
-    // TODO maybe change this to like "direction" or something so
-    // can support bottom, right, left etc...
-    // ie it gets that as string and transfers prop to menu?...
     'bottom': {
+      type: Boolean,
       default: false
     },
     'large': {
+      type: Boolean,
       default: false
     },
     'button_message': {
-      default: null,
-      type: String
+      type: String,
+      default: null
     },
     'width':{
+      type: [String, Number],
       default: undefined,
-      type: String,
     },
     'height':{
-      default: undefined,
-      type: String,
+      type: [String, Number],
+      default: undefined
     },
     'datacy':{
+      type: String,
       default: 'tooltip-button'
      },
     'target': {
         type: String,
         default: '_self'
      },
-    'ui_schema_name': {
-        type: String,
-        default: undefined
-     }
-  },
-  data() {
-    return {
-      // because we can't override props
-      // and rather have the interface be the "cleaner / shorter" thing
-      top_actual: true,
-      bottom_actual: false,
-      visible: true
-    }
   },
-  created(){
-    // defaults tp top, so setting bottom to true adds this?
-    // not quite right but should work
-    if (this.bottom == true) {
-      this.bottom_actual = true
-      this.top_actual = false
-    }
-    this.refresh_button_state_from_ui_schema()
-  },
-  mounted(){
-    this.show_ui_schema_refresh = this.$store.watch((state) => {
-        return this.$store.state.ui_schema.refresh
-      },
-      (new_val, old_val) => {
-        this.refresh_button_state_from_ui_schema()
-      },
-    )
-  },
-  beforeDestroy() {
-    this.show_ui_schema_refresh()
-  },
-  methods: {
-    refresh_button_state_from_ui_schema(){
-      if (this.$props.ui_schema_name == undefined) { return true }
-      this.visible = this.$store.getters.get_ui_schema(this.$props.ui_schema_name, 'visible')
-    },
-    preventdefault(event) {
-      // we don't assume this will go anywhere unless opened in new tab
-      // in which case the event will follow the default. In the future
-      // could offer an "open in new tab" specific new
-
-      if (this.target != '_self'){ return } // eg '_blank' we want default opening in new tab
```

Comment:
80% sure need this for opening in new tab (control click)
it's badly worded though, could maybe call it that directly and just do == _blank instead of the != which is confusing (I wrote it so just saying on myself)

Reply:
So, for now it was not used in the component, and I'd say if we don't use something maybe there is no sense to keep that for now?

Reply:
L12 on prior it was used 
https://github.com/diffgram/diffgram/pull/1091/files/ce626834e2ad7c2808656a70b664a07bfd4f1422#diff-647d454dbc5ac5b6868117c257fe4f2173ee33d40ac018f273df99f567f1d69eL12

Reply:
Sorry, my bad, lemme roll that back